bundles/org.openhab.core/src/main/java/org/openhab/core/util/ColorUtil.java

```diff
@@ -140,6 +142,93 @@ public static PercentType[] hsbToRgbPercent(HSBType hsb) {
+     *
+     * This function does not round the components. For conversion to integer values in the range 0 to 255 use
+     * {@link #hsbToRgb(HSBType)}.
+     *
+     * See also: {@link #hsbToRgb(HSBType)}, {@link #hsbTosRgb(HSBType)}, {@link #hsbToRgbPercent(HSBType)}
+     *
+     * @param hsb an {@link HSBType} value.
+     * @return array of four {@link PercentType} with the RGBW values in the range 0 to 100 percent.
+     */
+    public static PercentType[] hsbToRgbwPercent(HSBType hsb) {
+        PercentType[] rgb = hsbToRgbPercent(hsb);
+        final BigDecimal Ri = rgb[0].toBigDecimal();
+        final BigDecimal Gi = rgb[1].toBigDecimal();
+        final BigDecimal Bi = rgb[2].toBigDecimal();
+        // Get the maximum between R, G, and B
+        final BigDecimal tM = Ri.max(Gi.max(Bi));
+
+        // If the maximum value is 0, immediately return pure black.
+        if (tM.floatValue() == 0) {
+            return new PercentType[] { PercentType.ZERO, PercentType.ZERO, PercentType.ZERO, PercentType.ZERO };
+        }
+
+        // This section serves to figure out what the color with 100% hue is
+        final BigDecimal multiplier = BIG_DECIMAL_100.divide(tM, 0, RoundingMode.DOWN);
+        final BigDecimal hR = Ri.multiply(multiplier);
+        final BigDecimal hG = Gi.multiply(multiplier);
+        final BigDecimal hB = Bi.multiply(multiplier);
+
+        // This calculates the Whiteness (not strictly speaking Luminance) of the color
+        final BigDecimal M = hR.max(hG.max(hB));
+        final BigDecimal m = hR.min(hG.min(hB));
+        final BigDecimal Luminance = ((M.add(m).divide(BIG_DECIMAL_2).subtract(BIG_DECIMAL_50))
+                .multiply(BIG_DECIMAL_100.divide(BIG_DECIMAL_50))).divide(multiplier);
+
+        // check range
+        final BigDecimal Ro = Ri.subtract(Luminance).max(BigDecimal.ZERO);
+        final BigDecimal Go = Gi.subtract(Luminance).max(BigDecimal.ZERO);
+        final BigDecimal Bo = Bi.subtract(Luminance).max(BigDecimal.ZERO);
+        final BigDecimal Wo = Luminance.max(BigDecimal.ZERO);
+
+        return new PercentType[] { new PercentType(Ro), new PercentType(Go), new PercentType(Bo), new PercentType(Wo) };
+    }
+
+    /**
+     * Transform <a href="https://en.wikipedia.org/wiki/HSL_and_HSV">HSV</a> based {@link HSBType} to RGBW.
+     *
+     * See <a href=
+     * "https://stackoverflow.com/questions/40312216/converting-rgb-to-rgbw">Converting RGB to RGBW</a>.
+     *
+     * This function does not round the components. For conversion to integer values in the range 0 to 255 use
+     * {@link #hsbToRgb(HSBType)}.
+     *
+     * See also: {@link #hsbToRgb(HSBType)}, {@link #hsbTosRgb(HSBType)}, {@link #hsbToRgbPercent(HSBType)}
+     *
+     * @param rgb array of three int with the RGB values in the range 0 to 255.
+     * @return hsb an {@link HSBType} value.
+     *
+     */
+    public static HSBType rgbwTohsb(double r, double g, double b, double w) {
```

Comment:
We have used arrays of int or PercentType until now as input parameters. Existing method ```rgbToHsb(int[3])``` vs. new method ```rgbwToHsb(r,g,b,w)```. Do we need the precision of double, or would a ```rgbwToHsb(int[4])``` be sufficient (then IMHO there should be only one public interface which can handle arrays of size 3 and 4, maybe implemented by 2 private methods).